src/eredis.erl

```diff
@@ -48,8 +48,10 @@ start_link() ->
 %% socket as `{local, Path}' (available in OTP 19+); default `"127.0.0.1"'</dd>
 %% <dt>`{port, Port}'</dt><dd>Integer, default is 6379</dd>
 %% <dt>`{database, Database}'</dt><dd>Integer; 0 for the default database</dd>
-%% <dt>`{username, Username}'</dt><dd>String; default: no username</dd>
-%% <dt>`{password, Password}'</dt><dd>String; default: no password</dd>
+%% <dt>`{username, Username}'</dt><dd>String or a 0-ary function that returns
+%% the username; default: no username</dd>
```

Comment:
Or the atom `undefined` right? Also maybe it's a good idea to phrase it like this:

> the atom `undefined` for no username or a 0-ary wrapping function that returns the username; default is `undefined`;  a value of type `iodata()` is also accepted but discouraged (the 0-ary function is preferred as the passed value may be logged in erlang reports)

Reply:
Thx. The most important information should be first IMO so I put the fun first and `undefined` last.

> A 0-ary function that returns the username (the preferred way to provide username as it prevent the actual username from appearing in logs and stacktraces), a string or iodata or the atom `undefined` for no username; default is `undefined`